lib/Conversion/CombToAIG/CombToAIG.cpp

```diff
@@ -428,9 +442,204 @@ struct CombAddOpConversion : OpConversionPattern<AddOp> {
       carry = rewriter.create<comb::OrOp>(op.getLoc(),
                                           ValueRange{andOp, nextCarry}, true);
     }
+    LLVM_DEBUG(llvm::dbgs() << "Lower comb.add to Ripple-Carry Adder of width "
+                            << width << "\n");
 
     rewriter.replaceOpWithNewOp<comb::ConcatOp>(op, results);
-    return success();
+  }
+
+  // Implement a parallel prefix adder - with Kogge-Stone or Brent-Kung trees
+  // Will introduce unused signals for the carry bits but these will be removed
+  // by the AIG pass.
+  void LowerParallelPrefixAdder(comb::AddOp op, ValueRange inputs,
+                                ConversionPatternRewriter &rewriter) const {
+    auto width = op.getType().getIntOrFloatBitWidth();
+
+    auto aBits = extractBits(rewriter, inputs[0]);
+    auto bBits = extractBits(rewriter, inputs[1]);
+    // Construct propagate (p) and generate (g) signals
+    SmallVector<Value> p, g;
+
+    for (auto [aBit, bBit] : llvm::zip(aBits, bBits)) {
+      // p_i = a_i XOR b_i
+      p.push_back(rewriter.create<comb::XorOp>(op.getLoc(), aBit, bBit));
+      // g_i = a_i AND b_i
+      g.push_back(rewriter.create<comb::AndOp>(op.getLoc(), aBit, bBit));
+    }
+
+    LLVM_DEBUG(llvm::dbgs()
+               << "Lower comb.add to Parallel-Prefix of width " << width << "\n"
+               << "--------------------------------------- Init\n");
+
+    for (int64_t i = 0; i < width; ++i) {
+      // p_i = a_i XOR b_i
+      LLVM_DEBUG(llvm::dbgs()
+                 << "P0" << i << " = A" << i << " XOR B" << i << "\n");
+      // g_i = a_i AND b_i
+      LLVM_DEBUG(llvm::dbgs()
+                 << "G0" << i << " = A" << i << " AND B" << i << "\n");
+    }
+
```

Comment:
nit: enclose entire for-loop with debug
```suggestion
LLVM_DEBUG ({
    for (int64_t i = 0; i < width; ++i) {
      // p_i = a_i XOR b_i
      llvm::dbgs() << "P0" << i << " = A" << i << " XOR B" << i << "\n";
      // g_i = a_i AND b_i
     llvm::dbgs() << "G0" << i << " = A" << i << " AND B" << i << "\n"
    }
   });

```